### DISCUSSION TITLE
Avoid duplication in `~/.zsh_history`

### DISCUSSION THREAD

**Discussion by Liiaming** [ohmyzsh/ohmyzsh#11795] [Help] [answered]
There is a lot of duplicate history in `~/.zsh_history`, how can I avoid this?

I noticed that `lib/history.zsh` might be useful, but I tried copying the contents of this file into `~/.zshrc` or adding in `~/.zshrc`: `source $HOME/.oh -my-zsh/lib/history.zsh`, neither works

what should I do

**Comment by Liiaming** [comment]
Append `~/.zshrc`:
```shell
if [[ -r "${XDG_CACHE_HOME:-$HOME/.cache}/p10k-instant-prompt-${(%):-%n}.zsh" ]]; then
  source "${XDG_CACHE_HOME:-$HOME/.cache}/p10k-instant-prompt-${(%):-%n}.zsh"
fi

export ZSH="$HOME/.oh-my-zsh"

ZSH_THEME="powerlevel10k/powerlevel10k"

HIST_STAMPS="yyyy-mm-dd"

plugins=(
    sudo
    zsh-autosuggestions
    zsh-syntax-highlighting
)

source $ZSH/oh-my-zsh.sh

setopt no_nomatch

export NVM_DIR="$HOME/.nvm"
[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
[ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"  # This loads nvm bash_completion

export PATH="$HOME/.jenv/bin:$PATH"
eval "$(jenv init -)"

[[ ! -f ~/.p10k.zsh ]] || source ~/.p10k.zsh
```

**Comment by carlosala** [comment, marked as answer]
If you don't want to have dups in your history, I recommend adding `setopt hist_ignore_all_dups`. You can read the docs as well [here](https://zsh.sourceforge.io/Doc/Release/Options.html#History).
